### ISSUE TITLE
Issue with using date format strings

### ISSUE THREAD

**Issue by ericklind** [ericelliott/riteway#168]
I'm testing a formatting library, which formats values from Excel formats. When I'm trying to test the date component, I keep getting this error:
```
error fired SyntaxError: missing ) after argument list
        operator: deepEqual
        diff: "10.09.2019" => "09/Di/yyyy"
```
The function that I'm trying to test looks like this:

```
formatLocaleDate: (date, locale, format) => {
        let f = format || 'L';
        return moment(date, self.dateInputFormats, locale).format(f);
    }
```
My test function looks like this:
```
 assert({
        given: '2019-09-10T00:00:00.000-07:00',
         'return correct date for the locale.',
        actual: formatLocaleDate('2019-09-10T00:00:00.000-07:00', 'de', 'MM/dd/yyyy'),
        expected: '10.09.2019'
    });
```
Given:
- self.dateInputFormats is an array of moment constants.

If I remove the format param ('MM/dd/yyyy') the test works just fine. If I have it in there, you can see that it creates some kind of weird value: "09/Di/yyyy"

Any idea how to get past this or why it's trying to do something with the format, which is just an input string? Are the some escape sequences that I need to know about? 
I need to be able of pass all kinds template strings: #,##0.00, #,##0 and the like.

**Comment by ericelliott** [comment]
```js
assert({
        given: '2019-09-10T00:00:00.000-07:00',
         'return correct date for the locale.',
        actual: formatLocaleDate('2019-09-10T00:00:00.000-07:00', 'de', 'MM/dd/yyyy'),
        expected: '10.09.2019'
});
```

Where's the `should` part of the assertion?

**Comment by ericelliott** [comment]
> Don't worry about the syntax error. That's just what happens in RITEway when actual doesn't match expected.

I don't think that's accurate, and it's definitely not expected behavior - but I've never seen a syntax error reported by RITEway that had anything to do with RITEway. Generally it's because something is wrong with the test or the code under test.

**Comment by ericklind** [comment]
Hey,
I was able to get it working. I honestly don't remember what caused the issue, but it's working very smoothly. Thanks for creating this and pointing out that tests should be pretty simple and straight forward. All the testing examples I checked out before were so confusing that I pretty much just gave up on writing test. Once you pointed out "They either return what I expect or they don't" and testing for things in a functional style, it just clicked. I ended up writing about 80 tests for a particular library and was able to debug it so much faster than I would have been able to do without them.

Cheers!
